src/__tests__/three-idx.test.ts

```diff
@@ -0,0 +1,312 @@
+import { publishIDXConfig } from '@ceramicstudio/idx-tools'
+import { randomBytes } from '@stablelib/random'
+import ThreeIdResolver from '@ceramicnetwork/3id-did-resolver'
+
+import { NewAuthEntry, ThreeIDX } from '../three-idx'
+import { DidProvider } from '../did-provider'
+import Keyring from '../keyring'
+
+import dagJose from 'dag-jose'
+import { sha256 } from 'multiformats/hashes/sha2'
+import legacy from 'multiformats/legacy'
+import * as u8a from 'uint8arrays'
+import type { Hasher } from 'multiformats/hashes/hasher'
+import Permissions from '../permissions'
+import KeyDidResolver from 'key-did-resolver'
+
+const seed = u8a.fromString(
+  '8e641c0dc77f6916cc7f743dad774cdf9f6f7bcb880b11395149dd878377cd398650bbfd4607962b49953c87da4d7f3ff247ed734b06f96bdd69479377bc612b',
+  'base16'
+)
+const KEYCHAIN_DEF = definitions.threeIdKeychain
+
+const genIpfsConf = (folder: string): any => {
+  const hasher: Record<number, Hasher<string, number>> = {}
+  hasher[sha256.code] = sha256
+  const format = legacy(dagJose, { hashes: hasher })
+  return {
+    ipld: { formats: [format] },
+    repo: `${folder}/ipfs/`,
+    config: {
+      Addresses: { Swarm: [] },
+      Bootstrap: [],
+    },
+    silent: true,
+  }
+}
+const randomSecret = () => '0x' + Buffer.from(randomBytes(32)).toString('hex')
+
+const pauseSeconds = (sec: number) => new Promise((res) => setTimeout(res, sec * 1000))
+
+const fakeJWE = () => ({
+  jwe: {
+    protected: 'prot',
+    tag: 'tag',
+    ciphertext: randomSecret(),
+    iv: 'iv',
+  },
+})
+
+async function genAuthEntryCreate(): Promise<NewAuthEntry> {
+  const provider = new Ed25519Provider(randomBytes(32))
+  const did = new DID({ provider, resolver: KeyResolver.getResolver() })
+  await did.authenticate()
+  return {
+    did,
+    mapEntry: {
+      [did.id]: {
+        data: fakeJWE(),
+        id: fakeJWE(),
+      },
+    },
+  }
+}
+
+const setup3id = async (threeIdx: ThreeIDX, keyring: Keyring) => {
+  const genState = keyring.get3idState(true)
+  const forcedDID = genState.metadata.controllers[0]
+  let didProvider = new DidProvider({
+    permissions: mockedPermissions,
+    threeIdx,
+    keyring,
+    forcedDID,
+  })
+  await threeIdx.setDIDProvider(didProvider)
+  await threeIdx.create3idDoc(genState)
+  didProvider = new DidProvider({ permissions: mockedPermissions, threeIdx, keyring })
+  await threeIdx.setDIDProvider(didProvider)
+}
+
+const mockedPermissions = ({
+  request: () => Promise.resolve([]),
+  has: () => true,
+} as unknown) as Permissions
+
+jest.setTimeout(250000)
+let tmpFolder: DirectoryResult
+let ipfs: Ipfs.IPFS
+let ceramic: Ceramic
+let keyring: Keyring
+let threeIdx: ThreeIDX
+let anchorService: any
+
+beforeAll(async () => {
+  tmpFolder = await tmp.dir({ unsafeCleanup: true })
+  ipfs = await Ipfs.create(genIpfsConf(tmpFolder.path))
+  ceramic = await Ceramic.create(ipfs, {
+    stateStoreDirectory: tmpFolder.path + '/ceramic/',
+    anchorOnRequest: false,
+  })
```

Comment:
We could maybe use `jest-environment-ceramic` here instead of adding all of the setup code in this test?

Reply:
Most probably yes.